src/platform/datapath_winkernel.c

```diff
@@ -2373,12 +2319,10 @@ CxPlatDataPathSocketReceive(
         }
     }
 
-    if (ReleaseChain != NULL) {
-        //
-        // Release any dropped or copied datagrams.
-        //
-        Binding->DgrmSocket->Dispatch->WskRelease(Binding->Socket, ReleaseChain);
-    }
+    //
+    // Release any dropped or copied datagrams.
+    //
+    Binding->DgrmSocket->Dispatch->WskRelease(Binding->Socket, DataIndicationHead);
```

Comment:
Should/can we just return STATUS_SUCCESS at the end to indicate inline processing and then not call `WskRelease` at all?

Reply:
> A WSK application calls the WskRelease function to release a linked list of [WSK_DATAGRAM_INDICATION](https://learn.microsoft.com/en-us/windows-hardware/drivers/ddi/wsk/ns-wsk-_wsk_datagram_indication) structures that it previously retained by returning STATUS_PENDING from either its [WskReceiveEvent](https://learn.microsoft.com/en-us/windows-hardware/drivers/ddi/wsk/nc-wsk-pfn_wsk_receive_event) or its [WskReceiveFromEvent](https://learn.microsoft.com/en-us/windows-hardware/drivers/ddi/wsk/nc-wsk-pfn_wsk_receive_from_event) event callback function.

From the docs, return success and do not call release seems simplest.

Reply:
Updated.